woocommerce/wc-admin
# Update comparison chart design and show it on the Product report

Part of #695.

This PR does several things:
- Renders the charts with the comparison layout in the Product report.
- Updates the comparison layout to the new designs.
- Splits the CSS of the chart component in two different files (created a new one for the legend).

### Screenshots
![image](https://user-images.githubusercontent.com/3616980/48348528-37eac780-e647-11e8-8bd2-11a1d3218498.png)

### Detailed test instructions:
Given that the API endpoints are not ready yet, we must do a couple of tweaks in the code so we generate a long legend for the chart.

Go to `client/components/chart/legend.js` and replace [this line](https://github.com/woocommerce/wc-admin/compare/add/product-detail-report-chart?expand=1#diff-979f811d595d5bbcc243fc535a1ba3f0R83):
```ES6
const showTotalLabel = legendDirection === 'column' && data.length > 5 && itemsLabel;
```
with:
```ES6
const showTotalLabel = legendDirection === 'column' && data.length > 0 && itemsLabel;
```
and [this other line](https://github.com/woocommerce/wc-admin/compare/add/product-detail-report-chart?expand=1#diff-979f811d595d5bbcc243fc535a1ba3f0R102):
```ES6
{ data.map( row => ( 
```
with something like this:
```ES6
{ data.concat( data ).concat( data ).concat( data ).concat( data ).concat( data ).concat( data ).map( row => (
```
This way we will have many items in the legend and will be ready to test the PR.
- Go to the _Products_ report (`/wp-admin/admin.php?page=wc-admin#/analytics/products`).
- Verify the chart is displayed with the comparison layout (screenshot above):
  - The legend is on the left on wide screens and moves down to the bottom on mobile.
  - A shadow is displayed over the legend to make it clear it's scrollable and it disappears when you reach the end of the list.
  - A label is displayed at the bottom with the words _2 products_.
- Activate the Single Product filter (see screenshot).
![image](https://user-images.githubusercontent.com/3616980/48348773-e262ea80-e647-11e8-9a66-10ed50d7b33a.png)
  - Verify the label at the bottom of the legend changed to _2 variations_.

### To do
- The chart still displays all products data together instead of split by product/variation. I will leave #695 open to track that once the API allows querying the data that way.
- The chart height and the legend height don't match (as can be seen in the screenshot above). I will work on that in a follow-up PR, given that this one is already quite big.
